### PR TITLE
Upgrade to boost 1.80

### PR DESCRIPTION
**Pull request by JunielKatarn**
## Description
Upgrade Boost Framework dependency to version 1.80.

### Why
Adds ASIO and Beast (networking) security fixes and performance improvements.
See:
- https://www.boost.org/users/history/version_1_77_0.html
- https://www.boost.org/users/history/version_1_78_0.html
- https://www.boost.org/users/history/version_1_79_0.html
- https://www.boost.org/users/history/version_1_80_0.html

### What
PackageReference entries for `boost` get bumped from `1.76.0.0` to `1.80.0.0`.

## Testing
Pull Request CI validation.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/10533)

### CODE REVIEW

File: packages/playground/windows/playground-win32/Playground-win32.vcxproj [hunk 1]
```diff
@@ -169,7 +169,7 @@
     <Import Project="$(ReactNativeWindowsDir)\PropertySheets\External\Microsoft.ReactNative.Uwp.CppApp.targets" Condition="Exists('$(ReactNativeWindowsDir)\PropertySheets\External\Microsoft.ReactNative.Uwp.CppApp.targets')" />
   </ImportGroup>
   <ItemGroup>
-    <PackageReference Include="boost" Version="1.76.0.0" />
```
Comment (aeulitz) [inline packages/playground/windows/playground-win32/Playground-win32.vcxproj:172]: ><PackageReference Include="boost" Version="1.76.0.0" /> [](http://example.com/codeflow?start=4&length=55)

Does the UWP Playground app need a similar update, or does it use a different network stack?

File: packages/sample-apps/windows/SampleAppCPP/SampleAppCpp.vcxproj [hunk 2]
```diff
@@ -177,7 +177,7 @@
     </Page>
   </ItemGroup>
   <ItemGroup>
-    <PackageReference Include="boost" Version="1.76.0.0" />
+    <PackageReference Include="boost" Version="1.80.0.0" />
```
Comment (aeulitz) [inline packages/sample-apps/windows/SampleAppCPP/SampleAppCpp.vcxproj:180]: >"1.80.0.0" [](http://example.com/codeflow?start=46&length=10)

Given the sizeable number of .vcxproj files this appears in, should this version number be contained in a shared .props file?

**Review by aeulitz — APPROVED** [review]
:shipit:

File: packages/playground/windows/playground-win32/Playground-win32.vcxproj (hunk 1 above)
Comment (JunielKatarn) [reply packages/playground/windows/playground-win32/Playground-win32.vcxproj:172]: Both UWP and Desktop use the same stack.

File: packages/sample-apps/windows/SampleAppCPP/SampleAppCpp.vcxproj (hunk 2 above)
Comment (JunielKatarn) [reply packages/sample-apps/windows/SampleAppCPP/SampleAppCpp.vcxproj:180]: That's a pending task for most dependencies.

**Pull request closed without merging**